### ISSUE TITLE
Package is triggering autoload errors

### ISSUE THREAD

**Issue by jaapio** [zfcampus/zf-asset-manager#4]
For some reason our modules are triggering class not found exceptions when this package is added. It looks like this is only triggered when installing packages from source (git)

```


  [Symfony\Component\Debug\Exception\FatalThrowableError]
  Class '<Redacted>\UserModule\Module' not found


Exception trace:
 () at /data/vendor/<redacted>/user-module/config/user.config.php:53
 include() at /data/vendor/<redacted>/user-module/config/module.config.php:23
 include() at /data/vendor/zfcampus/zf-asset-manager/src/AssetInstaller.php:74
 ZF\AssetManager\AssetInstaller->__invoke() at /data/vendor/zfcampus/zf-asset-manager/src/Plugin.php:65
 ZF\AssetManager\Plugin->onPostPackageInstall() at n/a:n/a
 call_user_func() at phar:///usr/bin/composer/src/Composer/EventDispatcher/EventDispatcher.php:171
 Composer\EventDispatcher\EventDispatcher->doDispatch() at phar:///usr/bin/composer/src/Composer/EventDispatcher/EventDispatcher.php:116
 Composer\EventDispatcher\EventDispatcher->dispatchPackageEvent() at phar:///usr/bin/composer/src/Composer/Installer.php:605
 Composer\Installer->doInstall() at phar:///usr/bin/composer/src/Composer/Installer.php:223
 Composer\Installer->run() at phar:///usr/bin/composer/src/Composer/Command/InstallCommand.php:119
 Composer\Command\InstallCommand->execute() at phar:///usr/bin/composer/vendor/symfony/console/Command/Command.php:267
 Symfony\Component\Console\Command\Command->run() at phar:///usr/bin/composer/vendor/symfony/console/Application.php:846
 Symfony\Component\Console\Application->doRunCommand() at phar:///usr/bin/composer/vendor/symfony/console/Application.php:191
 Symfony\Component\Console\Application->doRun() at phar:///usr/bin/composer/src/Composer/Console/Application.php:227
 Composer\Console\Application->doRun() at phar:///usr/bin/composer/vendor/symfony/console/Application.php:122
 Symfony\Component\Console\Application->run() at phar:///usr/bin/composer/src/Composer/Console/Application.php:100
 Composer\Console\Application->run() at phar:///usr/bin/composer/bin/composer:54
 require() at /usr/bin/composer:24

````

**Comment by jaapio** [comment]
Think this is triggered by:
https://github.com/zfcampus/zf-asset-manager/blob/b580ac37f75141681a0c3d8565d8093906d22a57/src/AssetInstaller.php#L74

Since we are using ::class notation in our configs this will trigger errors.

**Comment by weierophinney** [comment]
Found a reproduce case in https://github.com/doctrine/DoctrineMongoODMModule/blob/master/config/module.config.php

You're correct in identifying the location of the error. The problem, however, will be mitigating it. The error raised is an `E_FATAL`, which is not something a userland error handler can catch.

I'll see if we can do some pre-analysis of the file before we attempt to include it.

**Comment by weierophinney** [comment]
Okay, I have a fix in #5 that should cover these issues.

The approach I took was to use `file_get_contents()` on the configuration file and pass it to `token_get_all()`. I then loop over each token, looking for problematic tokens (class/interface/trait definitions, `T_DOUBLE_COLON`, exits, evals, and one or two others. If any are found, I skip the file, and emit a warning.

This means that you cannot use `::class` notation in the same configuration that includes asset manager configuration, unfortunately. I cannot figure out a way around that due to the order-of-operations that would lead to being able to autoload classes. However, considering that this is primarily to allow aggregating assets related to the admin UI itself, I think we can likely live with that.

**Comment by weierophinney** [comment]
Fixed with #5, and released with 1.2.0.

**Comment by jaapio** [comment]
I agree with you that this solves this issue. Aslong the Admin Ui will not contain any `::class` notations. 
Thanks for this fix!
